### PR TITLE
Improvements to the chatbox

### PR DESCRIPTION
**Pull request by Excellify**
Different improvements to chat:

- css changes
- readable usernames
- message tokenizer rework
- mentions
- links 
- moderation slider
- mod actions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
 		"preview": "vite preview"
 	},
 	"dependencies": {
+		"color2k": "^2.0.0",
```
Comment (AnatoleAM) [inline package.json:20]: This should be a dev dependency

File: src/site/twitch.tv/modules/chat/components/BanSlider.vue [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	if (!tracking) return;
+	e.preventDefault();
+
+	const calcPos = Math.min(e.pageX - initial, maxVal);
+
+	data.value = new sliderData(calcPos);
+};
+</script>
+
+<style lang="scss">
```
Comment (AnatoleAM) [inline src/site/twitch.tv/modules/chat/components/BanSlider.vue:112]: This should be scoped

File: src/site/twitch.tv/index.ts [hunk 3]
```diff
@@ -109,3 +114,83 @@ export namespace Selectors {
 	export const ChatMessageUsername = ".chat-line__usernames";
 	export const ChatMessageTimestamp = ".chat-line__timestamp";
 }
+
+export const enum messageType {
```
[the remark this answers is not in the document]
Comment (AnatoleAM) [reply src/site/twitch.tv/index.ts:118]: Also the var name should be title case

[the remark this answers is not in the document]
Comment (Excellify) [reply src/site/twitch.tv/index.ts:118]: The enum values are literally just copied from twitch's files

[the remark this answers is not in the document]
Comment (AnatoleAM) [reply src/site/twitch.tv/index.ts:118]: oh LULE
well that's fine but do still format them properly kkona

**Pull request merged**